Copilot i18n docs (#3775)
vaadin/docs: articles/tools/copilot/i18n.adoc

---
title: Internationalization
description: Internationalization features in Copilot
order: 1
---

= [since:com.vaadin:vaadin@V24.5]#Internationalization#

> The Internationalization panel can be used to convert static strings used in the UI to "internationalized" translation API calls, generate translation keys and populate the default translation property file them.

Reviewer: Typo: 'them'

== Selecting Strings for Internationalization

Strings can be selected in two ways: by selecting individual UI element, or by clicking the [guibutton]*Find all static strings* button in the Internationalization panel, which automatically selects all elements containing static strings in the currently visible UI.

> In both cases, the elements are listed in the Internationalization panel. Each element is identified by the string inside it, and its HTML tag. These items can be expanded to reveal each string found in them. Those can be further expanded to reveal a proposed, automatically generated translation key, and the string that will be associated with it in the translation file. Checkboxes can be used to select or deselect entire elements or individual strings. The “All static strings” checkbox above the list can be used to select or deselect all items.

Reviewer: The two sentences with "These..." and "Those..." feel a bit disjointed and might confuse the reader.  In addition: How about adding an image?

By default, the list also shows elements and strings that are already internationalized, distinguished by a green checkmark symbol. These can be hidden by toggling the “Show internationalized strings” switch at the top.

== Modifying Proposed Keys and Associated Strings

Copilot automatically generates a translation key for each static string. You can modify the key by expanding the item's details and editing the Key field, or pick another existing key from the dropdown list.

> The string that will be associated with the key can also be edited. If an existing key is selected, the field where the associated string can be edited is locked by default to prevent accidentally modifying the string that is probably already used elsewhere. You can unlock it by clicking the button in the field.

Reviewer: Suggestions:
"string that is going to be associated" instead of "string that will be associated"
 "already in use" instead of "already used elsewhere".

== Applying the Current Selection

> The selected items are internationalized by clicking the [guibutton]*Apply* button at the bottom of the list. When the conversion is completed, a summary is displayed in the Internationalization panel.

Reviewer: Suggestion: Once the conversion is...

== Download Translation Files and Screenshots

Copilot also takes screenshots of the elements that it internationalizes. These can be used in the <<{articles}/control-center/localization#, Localization>> module in Control Center to provide translators with the context in which the strings are used.

> The screenshots are stored in the local storage of the browser that was used to run internationalization in Copilot. To download the screenshots, as well as the translation files in the project, expand the [guibutton]*Localization Data* panel below the [guibutton]*Find all static strings* button, and click the [guibutton]*Download Zip* button. This file can then be uploaded into the Localization module in Control Center, where localizations to different languages can be conveniently edited.

Reviewer: Maybe provide a screenshot for convenience.

== Selecting Strings for Internationalization

Strings can be selected in two ways: by manually selecting individual UI elements; or by clicking the [guibutton]*Find all static strings* button in the Internationalization panel. That automatically selects all elements containing static strings in the currently visible UI.

With both choices, the elements are listed in the Internationalization panel. Each element is identified by the string inside it -- and its HTML tag. These items can be expanded to reveal each string found in them. Those can be further expanded to reveal a proposed, automatically generated translation key, as well as the string that is to be associated with it in the translation file. Checkboxes can be used to select or deselect entire elements or individual strings. The "All static strings" checkbox above the list can be used to select or deselect all items.

By default, the list also shows elements and strings that are already internationalized, distinguished by a green checkmark symbol. Those internationalized elements can be hidden by toggling the "Show internationalized strings" switch at the top.

== Modifying Proposed Keys & Associated Strings

> Copilot automatically generates a translation key for each static string. You can modify the key by expanding the item's details and editing the Key field, or by pick another existing key from the drop-down list.

Author: Hmm, that also doesn't really work, since you need to expand before editing or picking. How about this

```suggestion
Copilot automatically generates a translation key for each static string. You can modify the key by expanding the item's details, where you can edit the Key field or pick an existing key from the drop-down list.
```
